packages/suite/src/reducers/wallet/coinjoinReducer.ts

```diff
@@ -85,13 +89,31 @@ const updateSession = (
     const account = draft.accounts.find(a => a.key === accountKey);
     if (!account || !account.session) return;
 
+    const { signedRounds, maxRounds, skipRounds } = account.session;
+    const { phase, phaseDeadline, roundDeadline } = round;
+
+    let sessionDeadline: string | number;
+    const roundsLeft = maxRounds - signedRounds?.length;
```

Comment:
```suggestion
    const roundsLeft = maxRounds - signedRounds.length - (typeof phase === 'number' ? 1 : 0)
```
- signedRounds is not optional, should be defined here
- exclude current round (if registered).  you know current round estimation from `roundDeadline` so you don't have to calculate it in getEstimatedTimePerRound
- sum sessionDeadlineRaw + timeLeftTillRoundEnd

Reply:
done